web/public/js/controllers/manage-dojo-users-controller.js

```diff
@@ -179,27 +176,34 @@ function cdManageDojoUsersCtrl($scope, $state, $q, cdDojoService, alertService,
       ngClick: function () {
         $scope.emailedUsers = []; // reset to allow animation to trigger while we load the new data set
         if ($scope.selectedItems.length !== 1) {
-          return cdDojoService.loadDojoUsers({dojoId: dojoId, userType: $scope.queryModel.userType, name: $scope.queryModel.name})
+
+          cdDojoService.loadDojoUsers({dojoId: dojoId, userType: $scope.queryModel.userType, name: $scope.queryModel.name})
           .then(function (users) {
+            $scope.emailedUsers = _.map(users.data.response, function (user) {
+              var response = {
+                userId: user.userId,
+                profileId: user.profileId,
+                name: user.name
+              };
+              return response;
+            });
             var kids = _.filter(users.data.response, function (user) {
               return _.includes(user.initUserType, 'attendee');
             });
-            async.mapSeries(kids, function (kid, cb) {
+            var parents = {};
+            async.each(kids, function (kid, cb) {
               cdUsersService.loadParentsForUserPromise(kid.id)
-              .then(function (parents) {
-                kid.parent = _.first(parents);
+              .then(function (kidParents) {
+                var parent = _.first(kidParents);
+                if (_.isArray(kidParents)) parents[kid.id] = {
+                  userId: parent.userId,
+                  profileId: parent.id,
+                  name: parent.name
+                };
                 cb();
               });
             }, function () {
-              $scope.emailedUsers = _.map(users.data.response, function (user) {
-                var response = {
-                  userId: user.userId,
-                  profileId: user.id,
-                  name: user.name
-                };
-                if (user.parent) response.parent = user.parent;
-                return response;
-              });
+              $scope.ccUsers = parents;
```

Comment:
no, i think email-users should be "relatively" independant (see contact dojo form) from manage-dojo-users